### ISSUE TITLE
[BUG] cannot create certificate signing request: Post https://xx/apis/certificates.k8s.io/v1beta1/certificatesigningrequests: x509: certificate signed by unknown authority

### ISSUE THREAD

**Issue by konghui** [openyurtio/openyurt#384]
**What happened**:
manual install yurthub  it display this error:
```
waiting for preparing client certificate
```
after solve this. it show
```
0625 11:10:21.580035       1 certificate_manager.go:434] Failed while requesting a signed certificate from the master: cannot create certificate signing request: Post https://xxx:6443/apis/certificates.k8s.io/v1beta1/certificatesigningrequests: x509: certificate signed by unknown authority 
```

**What you expected to happen**:
yurthub start correctly
**How to reproduce it (as minimally and precisely as possible)**:
kubectl apply -f config/setup/yurthub.yaml
**Anything else we need to know?**:

**Environment**:
- OpenYurt version:
- Kubernetes version (use `kubectl version`):
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**others**

/kind bug

**Comment by konghui** [comment]
Problem one is:  user should `--join-token=$bootstraptoken` on the yourthub.
Problem two is :If user chose manual install `openyurt` with self sign certificate. User should make sure `CA` certificate locate in  `/var/lib/yurthub/pki/ca.crt`.

**Comment by DreamEagle0** [comment]
Hi @konghui 
Since yurthub is deployed as static pod, I think you should not deploy it by running `kubectl apply -f` or change it to daemonset. Instead, you should put the yaml file in edgenode static pod path (default: /etc/kubernetes/manifests).

**Comment by konghui** [comment]
Ok thanks.

**Comment by rambohe-ch** [comment]
> Ok thanks. But I think if yurthub deployed as a static pod. It still need enough `RBAC` permissions

We have configured `RBAC` setting in default `config/setup/yurthub.yaml`. if you have any problems with yurthub, please let me in any way.

**Comment by DrmagicE** [comment]
Same issues here when I try to convert a pre-installed kubernetes cluster. It seams there were some problems for yurthub to get the certificate.
```
1 cert_mgr.go:398] avoid tcp conn leak, close old tcp conn that used to rotate certificate
I0908 13:52:08.426593       1 connrotation.go:110] forcibly close 1 connections on 192.168.33.220:6443 for hub certificate manager dialer
I0908 13:52:08.426771       1 connrotation.go:48] close connection from 192.168.33.221:35244 to 192.168.33.220:6443 for hub certificate manager dialer, remain 0 connections
I0908 13:52:08.429138       1 connrotation.go:145] create a connection from 192.168.33.221:35252 to 192.168.33.220:6443, total 1 connections in hub certificate manager dialer
E0908 13:52:08.436352       1 certificate_manager.go:434] Failed while requesting a signed certificate from the master: cannot create certificate signing request: Unauthorized
I0908 13:52:11.208252       1 certificate.go:83] waiting for preparing client certificate
I0908 13:52:12.749669       1 certificate_manager.go:409] Rotating certificates
I0908 13:52:16.209690       1 certificate.go:83] waiting for preparing client certificate
I0908 13:52:21.207619       1 certificate.go:83] waiting for preparing client certificate
```

Here is my command and output: 
```
[root@master-1 ~]# ./yurtctl convert --provider kubeadm --cloud-nodes master-1 -t -e  --kubeadm-conf-path /usr/lib/systemd/system/kubelet.service.d/10-kubeadm.conf --yurthub-healthcheck-timeout 5m
I0908 21:49:02.347153    1920 convert.go:318] mark master-1 as the cloud-node
I0908 21:50:12.451902    1920 util.go:492] servant job(yurtctl-disable-node-controller-master-1) has succeeded
I0908 21:50:12.451928    1920 convert.go:343] complete disabling node-controller
I0908 21:50:12.532002    1920 convert.go:354] yurt-tunnel-server is deployed
I0908 21:50:12.557186    1920 convert.go:362] yurt-tunnel-agent is deployed
I0908 21:50:12.578686    1920 convert.go:429] kube-public/cluster-info configmap already exists, skip to prepare it
CustomResourceDefinition/nodepools.apps.openyurt.io createdCustomResourceDefinition/uniteddeployments.apps.openyurt.io createdI0908 21:50:12.850090    1920 convert.go:381] yurt-app-manager is deployed
I0908 21:50:12.850108    1920 convert.go:385] deploying the yurt-hub and resetting the kubelet service...
E0908 21:52:12.877735    1920 util.go:489] fail to run servant job(yurtctl-servant-convert-worker-1): wait for job to be complete timeout
I0908 21:52:12.877988    1920 convert.go:409] complete deploying yurt-hub
```

**Comment by DrmagicE** [comment]
@rambohe-ch Maybe there are some problems with the bootstrap token.  The first one is generated by Yurthub,  the others are managed by kubeadm. I replaced the --join-token flags in yurthub.yaml with another token which is managed by kubeadm, and the Yurthub can get the certificate correctly. 
```
[root@master-1 ~]# kubeadm token list
TOKEN                     TTL         EXPIRES                     USAGES                   DESCRIPTION                                                EXTRA GROUPS
3ntkuv.cn8wjj1isz4ophfj   1h          2021-09-09T00:59:19+08:00   <none>                   Proxy for managing TTL for the kubeadm-certs secret        <none>
njpmre.ed6ot9gvw0p4w4m2   23h         2021-09-09T22:59:26+08:00   authentication,signing   <none>                                                     system:bootstrappers:kubeadm:default-node-token
s2ipqo.bvsvm2iu54b98hld   23h         2021-09-09T22:59:19+08:00   authentication,signing   The default bootstrap token generated by 'kubeadm init'.   system:bootstrappers:kubeadm:default-node-token
```

**Comment by rambohe-ch** [comment]
> Ok thanks. But I think if yurthub deployed as a static pod. It still need enough `RBAC` permissions

@konghui Please skip my former response, and you are right, we need add `RBAC` rights for yurthub. I will make a pull request to solve this problem.

**Comment by rambohe-ch** [comment]
> @rambohe-ch Maybe there are some problems with the bootstrap token. The first one is generated by Yurthub, the others are managed by kubeadm. I replaced the --join-token flags in yurthub.yaml with another token which is managed by kubeadm, and the Yurthub can get the certificate correctly.
> 
> ```
> [root@master-1 ~]# kubeadm token list
> TOKEN                     TTL         EXPIRES                     USAGES                   DESCRIPTION                                                EXTRA GROUPS
> 3ntkuv.cn8wjj1isz4ophfj   1h          2021-09-09T00:59:19+08:00   <none>                   Proxy for managing TTL for the kubeadm-certs secret        <none>
> njpmre.ed6ot9gvw0p4w4m2   23h         2021-09-09T22:59:26+08:00   authentication,signing   <none>                                                     system:bootstrappers:kubeadm:default-node-token
> s2ipqo.bvsvm2iu54b98hld   23h         2021-09-09T22:59:19+08:00   authentication,signing   The default bootstrap token generated by 'kubeadm init'.   system:bootstrappers:kubeadm:default-node-token
> ```

@DrmagicE Would you can parse these two kinds of bootstrap token, and show me the difference between them?

**Comment by rambohe-ch** [comment]
@konghui The problem is solved by pull request #510 
/close
